build.rs

```diff
@@ -213,45 +214,70 @@ fn main() {
     }
 }
 
-fn make_zlib(compiler: &cc::Tool, src_dir: &path::Path, out_dir: &path::Path) {
-    let src_dir = src_dir.join("zlib");
+fn make_zlib(compiler: &cc::Tool, src_dir: &path::Path, _: &path::Path) {
     // lock README such that if two crates are trying to compile
     // this at the same time (eg libbpf-rs libbpf-cargo)
     // they wont trample each other
-    let file = std::fs::File::open(src_dir.join("README")).unwrap();
+    let file = std::fs::File::open(src_dir.join("README.md")).unwrap();
     let _lock = fcntl::Flock::lock(file, fcntl::FlockArg::LockExclusive).unwrap();
 
-    let status = process::Command::new("./configure")
-        .arg("--static")
-        .arg("--prefix")
-        .arg(".")
-        .arg("--libdir")
-        .arg(out_dir)
-        .env("CC", compiler.path())
-        .env("CFLAGS", compiler.cflags_env())
-        .current_dir(&src_dir)
-        .status()
-        .expect("could not execute make");
-
-    assert!(status.success(), "make failed");
-
-    let status = process::Command::new("make")
-        .arg("install")
-        .arg("-j")
-        .arg(&format!("{}", num_cpus()))
-        .current_dir(&src_dir)
-        .status()
-        .expect("could not execute make");
-
-    assert!(status.success(), "make failed");
+    let zlib_sources = [
+        "adler32.c",
+        "compress.c",
+        "crc32.c",
+        "deflate.c",
+        "gzclose.c",
+        "gzlib.c",
+        "gzread.c",
+        "gzwrite.c",
+        "infback.c",
+        "inffast.c",
+        "inflate.c",
+        "inftrees.c",
+        "trees.c",
+        "uncompr.c",
+        "zutil.c",
+    ];
+
+    let cflags = [
+        // We do support hidden visibility, so turn that on.
+        "-DHAVE_HIDDEN",
+        // We do support const, so turn that on.
+        "-DZLIB_CONST",
+        // Enable -O3 as per chromium.
+        "-O3",
+        // "-Wall",
+        // "-Werror",
+        // "-Wno-deprecated-non-prototype",
+        // "-Wno-unused",
+        // "-Wno-unused-parameter",
```

Comment:
Where are these flags coming from, all of a sudden? And what does chromium have to do with this?

Reply:
In general, I kind of fail to understand what we gain from building our own build logic. The general guideline I am aware of is that the original build system [should be used](https://kornel.ski/rust-sys-crate#:~:text=Use%20the%20original%20build%20system%20of%20the%20library). If it provided a value, perhaps, but what value is that? Is the existing `zlib` build system incapable, somehow? Perhaps if you *finally* wrote a proper commit description I wouldn't have to inquire about every single detail and it would even be available for future developers...

Reply:
FYI Android keeps forks of elfutils(https://android.googlesource.com/platform/external/elfutils) and zlib(https://android.googlesource.com/platform/external/zlib/+/refs/heads/main) that uses the soong blueprint + ninja build system. 

Clearly some flags are copied from there: https://android.googlesource.com/platform/external/zlib/+/refs/heads/main/Android.bp#20 . (That Android.bp file doesn't have a license header.)

Some of the flags used in the soong blueprint are known to be supported by the compiler prebuilt that the AOSP project distributes, but might not be supported by all compilers. I don't think it is appropriate to set those flags for non Android targets.

Reply:
> FYI Android keeps forks of elfutils(https://android.googlesource.com/platform/external/elfutils) and zlib(https://android.googlesource.com/platform/external/zlib/+/refs/heads/main) that uses the soong blueprint + ninja build system.
> 
> Clearly some flags are copied from there: https://android.googlesource.com/platform/external/zlib/+/refs/heads/main/Android.bp#20 . (That Android.bp file doesn't have a license header.)
> 
> Some of the flags used in the soong blueprint are known to be supported by the compiler prebuilt that the AOSP project distributes, but might not be supported by all compilers. I don't think it is appropriate to set those flags for non Android targets.

Yes, these flags are only used in Android

Reply:
The larger question seems unanswered.

> In general, I kind of fail to understand what we gain from building our own build logic. The general guideline I am aware of is that the original build system [should be used](https://kornel.ski/rust-sys-crate#:~:text=Use%20the%20original%20build%20system%20of%20the%20library). If it provided a value, perhaps, but what value is that? Is the existing `zlib` build system incapable, somehow? Perhaps if you _finally_ wrote a proper commit description I wouldn't have to inquire about every single detail and it would even be available for future developers...

Putting that aside, please remove unused flags as well as references to Chromium. *Perhaps* these make some kind of sense in an Android build system context, but they have no business here. Also please make Android specific stuff guarded by `build_android()` (or whatever the name) as you do elsewhere.